packages/xo-web/src/common/render-xo-item.js

```diff
@@ -60,147 +108,206 @@ export const VmItem = decorate([
     }
   }),
-  ({ vm, container, ...props }) => (
-    <XoItem item={vm} to={`/vms/${get(() => vm.id)}`} {...props}>
-      {() => (
-        <span>
-          <Icon icon={`vm-${vm.power_state.toLowerCase()}`} />{' '}
-          {vm.name_label || vm.id}
-          {container !== undefined &&
-            ` (${container.name_label || container.id})`}
-        </span>
-      )}
-    </XoItem>
-  ),
+  ({ vm, container, link, newTab }) => {
+    if (vm === undefined) {
+      return UNKNOWN_ITEM
+    }
+
+    return (
+      <LinkWrapper link={link} newTab={newTab} to={`/vms/${vm.id}`}>
+        <Icon icon={`vm-${vm.power_state.toLowerCase()}`} /> {vm.name_label}
+        {container !== undefined && ` (${container.name_label})`}
+      </LinkWrapper>
+    )
+  },
 ])
 
-VmItem.propTypes = XO_ITEM_PROP_TYPES
+Vm.propTypes = {
+  id: PropTypes.string.isRequired,
+  link: PropTypes.bool,
+  newTab: PropTypes.bool,
+}
 
-export const SrItem = decorate([
+Vm.defaultProps = {
+  link: false,
+  newTab: false,
+}
+
+// ===================================================================
+
+export const VmTemplate = decorate([
   connectStore(() => {
-    const getSr = createGetObject()
-    return {
-      sr: getSr,
-      container: createGetObject(
-        createSelector(getSr, sr => get(() => sr.$container))
-      ),
-    }
+    const getObject = createGetObject()
+    return (state, props) => ({
+      // FIXME: props.self ugly workaround to get object as a self user
+      template: getObject(state, props, props.self),
+    })
   }),
-  ({ sr, container, ...props }) => (
-    <XoItem item={sr} to={`/srs/${get(() => sr.id)}`} {...props}>
-      {() => (
-        <span>
-          <Icon icon='sr' /> {sr.name_label || sr.id}
-          {container !== undefined && (
-            <span className='text-muted'> - {container.name_label}</span>
-          )}
-          {isSrWritable(sr) && (
-            <span>{` (${formatSize(sr.size - sr.physical_usage)} free)`}</span>
-          )}
-        </span>
-      )}
-    </XoItem>
-  ),
+  ({ template }) => {
+    if (template === undefined) {
+      return UNKNOWN_ITEM
+    }
+
+    return (
+      <span>
+        <Icon icon='vm' /> {template.name_label}
+      </span>
+    )
+  },
 ])
 
-SrItem.propTypes = XO_ITEM_PROP_TYPES
+VmTemplate.propTypes = {
+  id: PropTypes.string.isRequired,
+}
 
-export const RemoteItem = decorate([
-  addSubscriptions(({ id }) => ({
-    remote: cb =>
-      subscribeRemotes(remotes => {
-        cb(get(() => remotes.find(remote => remote.id === id)))
-      }),
-  })),
-  ({ remote, ...props }) => (
-    <XoItem item={remote} to='/settings/remotes' {...props}>
-      {() => (
-        <span>
-          <Icon icon='remote' /> {remote.name}
-        </span>
-      )}
-    </XoItem>
-  ),
+VmTemplate.defaultProps = {
+  link: false,
+  newTab: false,
+}
+
+// ===================================================================
+
+export const Sr = decorate([
+  connectStore(() => {
```

Comment:
```js
  connectStore(() => {
    const getSr = (state, props) => createGetObject()(state, props, props.self)
    return {
      sr: getSr,
      container: createGetObject(
        createSelector(
          getSr,
          (sr = {}) => sr.$container
        )
      ),
    }
  })

Reply:
`(state, props) => createGetObject()(state, props, props.self)` would create a new selector (thus invalidate the cache) on each call.